### ISSUE TITLE
Uncaught Error: Unable to find correct child when trying to convert JSON Schema to type

### ISSUE THREAD

**Issue by mbonaci** [mulesoft/api-workbench#347]
[Enter steps to reproduce below:]

1. Open RAML 0.8 with `urn:jsonschema` IDs - find example at the bottom of this issue
2. Right-click on any of `urn:jsonschema` and select `Refactoring > Convert JSON schema to type`

**Atom Version**: 1.11.2
**Electron Version**: 0.37.8
**System**: Mac OS X 10.11.6
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.40

### Stack Trace

Uncaught Error: Unable to find correct child

```
At /Users/mbo/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/mutators.js:370

Error: Unable to find correct child
    at Object.addToNode (/Users/mbo/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/mutators.js:370:19)
    at ASTNodeImpl.add (/Users/mbo/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/highLevelImpl.js:1392:18)
    at BasicSecuritySchemeImpl.BasicNodeImpl.addToProp (/Users/mbo/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/wrapped-ast/parserCore.js:139:26)
    at SchemaToModelGenerator.generateTo (/Users/mbo/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/tools/schemaModelGen.js:55:54)
    at e.performOk (/Users/mbo/.atom/packages/api-workbench/main.js:2:19074)
    at /Users/mbo/.atom/packages/api-workbench/main.js:2:16584
    at /Users/mbo/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:413:68
    at Array.forEach (native)
    at HTMLButtonElement.element.onclick (/Users/mbo/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:413:31)
```

### Commands

```
     -0:15.2.0 api-workbench:Convert JSON schema to type (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "react",
      "floobits",
      "language-babel"
    ],
    "telemetryConsent": "limited",
    "themes": [
      "one-dark-ui",
      "atom-dark-fusion-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.40 (active)
atom-yamljson, v0.2.3 (active)
floobits, v0.32.6 (inactive)
language-babel, v2.46.0 (inactive)
linter, v1.11.18 (active)
react, v0.16.1 (inactive)
atom-dark-syntax, v0.27.0 (active)
atom-dark-ui, v0.52.0 (active)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.2.0 (inactive)
base16-tomorrow-light-theme, v1.2.0 (inactive)
one-dark-ui, v1.6.0 (inactive)
one-light-ui, v1.6.0 (inactive)
one-dark-syntax, v1.3.0 (inactive)
one-light-syntax, v1.3.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.7.0 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.2 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.201.1 (active)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.59.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v1.0.0 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2-hotfix1 (active)
snippets, v1.0.2 (active)
spell-check, v0.68.2 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.101.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.209.3 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.33.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.2 (active)
language-csharp, v0.12.1 (active)
language-css, v0.37.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.42.1 (active)
language-html, v0.45.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.23.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.2 (active)
language-less, v0.29.5 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.2 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.69.0 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.56.0 (active)
language-shellscript, v0.22.4 (active)
language-source, v0.9.0 (active)
language-sql, v0.23.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.9 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

Generated with [springmvc-raml-plugin](https://github.com/phoenixnap/springmvc-raml-plugin)):

```yaml
#%RAML 0.8
title: new-ui
version: "0.0.1"
baseUri: /ua/api
/login: 
    displayName: Login Resource
    description: Login Resource
    post: 
        description: processLoginSubmit
        body: 
            application/json: 
                schema: |
                    {
                      "type" : "object",
                      "id" : "urn:jsonschema:com:sematext:ua:api:UserLoginDto",
                      "properties" : {
                        "loginName" : {
                          "type" : "string"
                        },
                        "password" : {
                          "type" : "string"
                        },
                        "rememberMe" : {
                          "type" : "boolean"
                        }
                      }
                    }
        responses: 
            "200": 
                description: Successful Response
                body: 
                    application/json: 
                        schema: |
                            {
                              "type" : "object",
                              "id" : "urn:jsonschema:com:sematext:ua:api:ApiResponse",
                              "properties" : {
                                "success" : {
                                  "type" : "boolean"
                                },
                                "message" : {
                                  "type" : "string"
                                },
                                "data" : {
                                  "type" : "object",
                                  "additionalProperties" : {
                                    "type" : "any"
                                  }
                                },
                                "errors" : {
                                  "type" : "array",
                                  "items" : {
                                    "type" : "object",
                                    "id" : "urn:jsonschema:com:sematext:ua:api:Error",
                                    "properties" : {
                                      "code" : {
                                        "type" : "string"
                                      },
                                      "message" : {
                                        "type" : "string"
                                      }
                                    }
                                  }
                                }
                              }
                            }
```

**Comment by ddenisenko** [comment]
The action should be disabled for RAML 0.8 as there are no types there.

**Comment by dreamflyer** [comment]
will be published in v0.8.42.
